ui/src/app/views/rename-modal/component.tsx

```diff
@@ -1,72 +1,117 @@
 import {Component, Input} from '@angular/core';
+import {validate} from 'validate.js';
 
 import {
   Button
 } from 'app/components/buttons';
 import {styles as headerStyles} from 'app/components/headers';
-import {TextInput} from 'app/components/inputs';
+import {TextInput, ValidationError} from 'app/components/inputs';
 import {
   Modal,
   ModalBody,
   ModalFooter,
   ModalTitle,
 } from 'app/components/modals';
+import {TooltipTrigger} from 'app/components/popups';
+import {workspacesApi} from 'app/services/swagger-fetch-clients';
 
-import {ReactWrapperBase} from 'app/utils';
+import {ReactWrapperBase, summarizeErrors} from 'app/utils';
 
 import * as React from 'react';
 
 
+const fullName = name => {
+  return !name || /^.+\.ipynb$/.test(name) ? name : `${name}.ipynb`;
+};
+
 interface RenameModalProps {
-  resource: { name: string };
+  resource: {name: string};
```

Comment:
i am kind of against passing in this pseudo-resource-- i was finding some difficulty with type stuff trying to give it a "full" resource (either notebook, cohort, or concept set).  if all we're using is the name, let's just pass `resourceName: string` as the prop.  I made this change for the confirm delete modal, it reduced some of the complexity in the modal itself

Reply:
Good point, it would be cleaner that way.